### PR TITLE
Add Overlapping Community Leaderboards to PSI Section

### PR DESCRIPTION
**Pull request by stevenelleman**
If there are shared communities AND top 10 leaderboard entries: 
Screenshot 2024-11-06 at 12 17 12 AM

If there are no shared communities OR no top 10 leaderboard entries: 
Screenshot 2024-11-06 at 12 16 15 AM

### CODE REVIEW

File: apps/frontend/src/pages/people/[username].tsx [hunk 1]
```diff
@@ -185,6 +196,22 @@ const CommentModal: React.FC<CommentModalProps> = ({
...
+  position: number,
+}
+
+type LeaderboardPositionRecord = Record<string, LeaderboardPosition[]>;
+
+const HumanReadableValues: Record<string, string> = {
+  "EDGE_CITY_LANNA": "Edge City Lanna",
+  "DEVCON": "Devcon",
+  "TOTAL_TAP_COUNT": "total tap count",
+  "STRAVA_PREVIOUS_MONTH_RUN_DISTANCE": "running distance over last month",
```
Comment (stevenelleman) [inline apps/frontend/src/pages/people/[username].tsx:210]: *distance running over last month

File: apps/frontend/src/pages/profile/overview.tsx [hunk 2]
```diff
@@ -62,11 +62,11 @@ export default function ProfileOverview() {
     >
       <div className="flex flex-col gap-3 mt-[46px]">
         <div className="flex flex-col px-4">
-          <span className="text-[30px] font-semibold tracking-[-0.22px] font-sans text-label-primary">
-            {user?.userData.displayName}
+          <span className="text-[30px] font-semibold tracking-[-0.22px] font-sans">
+            {`${user?.userData.username}`}
```
Comment (stevenelleman) [inline apps/frontend/src/pages/profile/overview.tsx:66]: In one location we had username at the top and display below it, and two cases with the reverse. Given that quite a few users have not filled out the display name, and username is required, I swapped the naming to consistently put username first and display name below it.

File: packages/types/src/chip/index.ts [hunk 3]
```diff
@@ -117,6 +117,19 @@ export enum LeaderboardEntryType {
...
+  LeaderboardEntryType.STRAVA_PREVIOUS_MONTH_RUN_DISTANCE,
+  LeaderboardEntryType.GITHUB_WEEK_OCT_20_COMMITS,
+];
+
+export const LannaLeaderboards: LeaderboardEntryType[] = [
+  LeaderboardEntryType.WEEK_OCT_20_TAP_COUNT,
+  LeaderboardEntryType.WEEK_OCT_27_TAP_COUNT,
+  LeaderboardEntryType.GITHUB_LANNA_COMMITS,
+  LeaderboardEntryType.LANNA_TOTAL_WORKOUT_COUNT,
+]
```
Comment (stevenelleman) [inline packages/types/src/chip/index.ts:131]: Going to sleep on this, whether it should be shared or not.

File: apps/frontend/src/pages/people/[username].tsx [hunk 4]
```diff
@@ -201,8 +228,55 @@ const UserProfilePage: React.FC = () => {
...
+    const fetchLeaderboardDetailsForCommunities = async () => {
+      if (verifiedIntersection?.communities && !gotLeaderboardPositions) {
+
+        // Only run computation once there's a community overlap
+        for (const issuer of verifiedIntersection?.communities) {
+          const community: ChipIssuer =  ChipIssuerSchema.parse(issuer);
+
+          let leaderboardTypes: LeaderboardEntryType[] = SharedLeaderboards;
+          switch(community) {
+            case ChipIssuer.EDGE_CITY_LANNA:
```
Comment (AndrewCLu) [inline apps/frontend/src/pages/people/[username].tsx:248]: do we have devcon leaderboards set up?

**Review by AndrewCLu — APPROVED** [review]
approved pending testing with lanna + devcon leaderboards

File: apps/frontend/src/pages/people/[username].tsx [hunk 5]
```diff
@@ -201,8 +228,55 @@ const UserProfilePage: React.FC = () => {
...
+  // NOTE: should leaderboard position state be added to useEffect dependency array?
+  useEffect(() => {
+    const fetchLeaderboardDetailsForCommunities = async () => {
+      if (verifiedIntersection?.communities && !gotLeaderboardPositions) {
+
+        // Only run computation once there's a community overlap
+        for (const issuer of verifiedIntersection?.communities) {
+          const community: ChipIssuer =  ChipIssuerSchema.parse(issuer);
+
+          let leaderboardTypes: LeaderboardEntryType[] = SharedLeaderboards;
```
[the remark this answers is not in the document]
Comment (stevenelleman) [reply apps/frontend/src/pages/people/[username].tsx:246]: Shared between the two users, I'll change name to "SharedIssuerLeaderboards"

File: apps/frontend/src/pages/people/[username].tsx [hunk 6]
```diff
@@ -554,8 +653,43 @@ const UserProfilePage: React.FC = () => {
...
+                      {/* After communities are ironed out add links to community page and / or dashboards */}
+                      {
+                        Object.keys(leaderboardPositions).map((community: string) => {
+                          const messages: string[] = [];
+                          const name: string = HumanReadableValues[community];
+
+                          for (const entry of leaderboardPositions[community]) {
+                            const board: string = HumanReadableValues[entry.type];
+
+                            const message = `#${entry.position} for ${board} at ${name}.`;
```
Comment (RiverRuby) [inline apps/frontend/src/pages/people/[username].tsx:672]: I want to clean up this copy + styling

**Review by RiverRuby — CHANGES_REQUESTED** [review]
Don't merge until I get to update copy+styling later

**Comment by RiverRuby** [comment]
Closing since this is outdated and unclear that we want this for now. Discovering overlapping communities won't be useful until we have many more communities.

**Comment by stevenelleman** [comment]
> Closing since this is outdated and unclear that we want this for now. Discovering overlapping communities won't be useful until we have many more communities.

@RiverRuby the point of this was to surface leaderboard on tap (ie if you're top 10 on github contributions, it's surfaced), but if you do that you need to handle the case where you have multiple communities. The multiple community aspect is secondary.

**Pull request closed without merging**